README.md

```diff
@@ -0,0 +1,47 @@
+# NJOY metaconfigure
+The metaconfigure package contains a suite of Python scripts to assist in the generation of build system configuration files for NJOY projects.
+
+## Updating metaconfigure in a project
+**Note:** that all the subprojects/dependencies must be updated prior to updating the super project.
+
+These commands (unless otherwise specified) should be run from the top level of a project.
+1. Create a branch
+2. Download ("fetch") subprojects
+   ```bash
+   python metaconfigure/fetch_subprojects.py
+   ```
+3. Update the metaconfigure subtree
+   ```bash
+   git subtree pull --squash --prefix metaconfigure https://github.com/njoy/metaconfigure.git master
+   ```
+4. Delete old `description.json` file.
+   ```bash
+   rm metaconfigure/description.json
+   ```
+5. Generate new `metaconfigure/description.json` file. This is done in Python.
+   ```python
+   import metaconfigure
+   args = {"name": "project-name", "language": "c++", "is external project": False, "include path": "src/include"}
+   metaconfigure.description.generate(**args)
+   ```
+   Note that the values in `args` need to be appropriate for your particular project. The values given above are just an example.
+6. Inspect the new `metaconfigure/description.json`. In particular check:
+	- `implementation files`—ensure the correct relative path (to top-level of project) is correct.
+	- `header files`—ensure the correct relative path (to top-level of project) is correct.
+	- `include path`—ensure that this points to the correct path. This is used for projects external to NJOY such as [catch-adapter](https://github.com/njoy/catch-adapter).
+7. Generate a [CMake](https://cmake.org) file
+   ```bash
+   metaconfigure/generate.py cmake
+   ```
+8. Inspect resulting `CMakelists.txt` file
+	 - Is the target of the expected type (executable, interface, or library)?
```

Comment:
Typically these appear near the end of the root source tree directory `CMakeLists.txt`. These types are described in the CMake documentation. An executable target will always be accompanied by a library or interface target in the CMake files generated by by metaconfigure, so we'll start there. 

A library target is a collection of source files which are to be compiled into a library file (either static or dynamic based on configuration settings) during the build process. This implies that the project contains one or more implementation files (excluding a implementation file for any executable driver). Those declarations look something like

```
add_library( <name> ${<name>_library_linkage} <source files> )
```

(How's that for meta?)

For targets that lack implementation files (such as header only libraries), there's nothing to compile. As a historical note, prior to CMake 3.x, the only way to describe these libraries was using `add_custom_target` and specify several (undocumented) properties. In modern CMake, these targets are described using a "interface library" (which feels very kludgy, but oh well). These declarations look like

```
add_library( <name> INTERFACE <source files> )
```

Given an driver implementation file (defaulting to main.<extension> but can be specified in the description), you should also see an executable target which links to the library or interface target.
```
add_executable( <name>_executable <driver> )
set_target_properties( <name>_executable PROPERTIES OUTPUT_NAME <name> )
target_compile_options( <name>_executable PRIVATE <flags generator expression> )
target_link_libraries( <name>_executable <PUBLIC/INTERFACE> <name> )
```